### PR TITLE
Provide resampled data bucket and table

### PR DESCRIPTION
**Pull request by galexite**
Create a resampled AIS bucket, associated Glue tables and crawler-related specifications, by refactoring out common crawler information in to a separate module.

### CODE REVIEW

File: deployment/src/main/terraform/modules/data-query/variables.tf [hunk 1]
```diff
@@ -1,11 +1,16 @@
-variable data_query_name {}
+variable ais_raw_catalog_database_name {}
```
Comment (JoelClemence) [inline deployment/src/main/terraform/modules/data-query/variables.tf]: Rename to be general as it isn't just raw data in the database

```suggestion
variable ais_catalog_database_name {}
```

File: deployment/src/main/terraform/modules/data-query/main.tf [hunk 2]
```diff
@@ -1,18 +1,8 @@
...
 }
 
-module catalog {
-  source                      = "./catalog"
-  catalog_database_name       = var.catalog_database_name
-  catalog_database_table_name = var.catalog_database_table_name
-  data_prefix                 = var.data_prefix
-  data_store_name             = var.data_store_name
+resource aws_glue_catalog_database catalog_database {
+  name = var.ais_raw_catalog_database_name
```
Comment (JoelClemence) [inline deployment/src/main/terraform/modules/data-query/main.tf]:
```suggestion
  name = var.ais_catalog_database_name
```

File: deployment/src/main/terraform/main.tf [hunk 3]
```diff
@@ -63,11 +63,13 @@ module notifications {
...
-  data_query_name               = data.external.secrets.result["data_query_name"]
-  data_query_results_store_name = data.external.secrets.result["data_query_results_store"]
-  catalog_database_name         = "ukho-ais-data"
-  catalog_database_table_name   = "raw_ais_data"
-  data_prefix                   = ""
-  data_store_name               = data.external.secrets.result["raw_partitioned_ais_store"]
+  source                                                = "./modules/data-query"
+  ais_raw_catalog_database_name                         = "ukho-ais-data"
+  ais_raw_catalog_database_table_name                   = "raw_ais_data"
+  ais_raw_partitioned_data_prefix                       = ""
```
Comment (JoelClemence) [inline deployment/src/main/terraform/main.tf]: Make ais_raw_catalog_database_table_name default to "" and remove

File: deployment/src/main/terraform/main.tf [hunk 4]
```diff
@@ -63,11 +63,13 @@ module notifications {
...
+  ais_raw_partitioned_store_name                        = data.external.secrets.result["raw_partitioned_ais_store"]
+  resampled_partitioned_ais_catalog_database_name       = "ukho-resampled-ais-data"
+  resampled_partitioned_ais_catalog_database_table_name = "resampled_ais_data"
+  resampled_partitioned_ais_data_prefix                 = ""
```
Comment (JoelClemence) [inline deployment/src/main/terraform/main.tf]: Ditto - make resampled_partitioned_ais_data_prefix default to "" and remove.

Probably worth removing this functionality as we don't need it.

File: deployment/src/main/terraform/main.tf [hunk 5]
```diff
@@ -21,11 +21,11 @@ module s3_buckets {
...
-  source                         = "./modules/storage"
-  data_query_results_store_name  = data.external.secrets.result["data_query_results_store"]
-  raw_ais_store_name             = data.external.secrets.result["raw_ais_store"]
-  ais_data_upload_store_name     = data.external.secrets.result["ais_data_upload_store"]
-  raw_partitioned_ais_store_name = data.external.secrets.result["raw_partitioned_ais_store"]
+  source                               = "./modules/storage"
+  raw_ais_store_name                   = data.external.secrets.result["raw_ais_store"]
+  ais_data_upload_store_name           = data.external.secrets.result["ais_data_upload_store"]
+  raw_partitioned_ais_store_name       = data.external.secrets.result["raw_partitioned_ais_store"]
+  resampled_partitioned_ais_store_name = data.external.secrets.result["resampled_partitioned_ais_store_name"]
```
Comment (andyphelps) [inline deployment/src/main/terraform/main.tf]: Inconsistent naming of the secret result?

```suggestion
  resampled_partitioned_ais_store_name = data.external.secrets.result["resampled_partitioned_ais_store"]
```

File: deployment/src/main/terraform/main.tf (hunk 3 above)
Comment (andyphelps) [reply deployment/src/main/terraform/main.tf]: Why?   The table needs a name?

Comment (andyphelps) [inline deployment/src/main/terraform/main.tf]: If this is defaulting to "" and isn't being reset anywhere it should probably be removed?

File: deployment/src/main/terraform/modules/data-query/crawler/main.tf [hunk 6]
```diff
@@ -0,0 +1,67 @@
+resource aws_iam_role catalog_database_crawler_execution_role {
```
Comment (andyphelps) [inline deployment/src/main/terraform/modules/data-query/crawler/main.tf]: What is the purpose of this crawler, couldn't find anything in the commit messages explaining why it has been added...

File: deployment/src/main/terraform/modules/data-query/table/main.tf [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+      serialization_library = "org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe"
+
+      parameters = {
+        "field.delim"          = "\t"
+        "mapkey.delim"         = "\u0003"
+        "collection.delim"     = "\u0002"
+        "serialization.format" = "\t"
+      }
+    }
+
```
Comment (andyphelps) [inline deployment/src/main/terraform/modules/data-query/table/main.tf]: Is this missing the partition config?

**Comment by galexite** [comment]
The partition configuration is included in #70 I believe (at
@JoelClemence's request), to help us get closer to the PBI. We have to
merge these files manually.

>

File: deployment/src/main/terraform/modules/data-query/table/main.tf [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+      type = "string"
+    }
+
+    columns {
+      name = "altitude"
+      type = "string"
+    }
+
+    columns {
+      name = "special_manoeurve"
```
Comment (oliverCausley) [inline deployment/src/main/terraform/modules/data-query/table/main.tf]: manoeuvre?

**Pull request merged**